rpcs3/rpcs3qt/settings_dialog.cpp

```diff
@@ -1042,6 +1039,43 @@ settings_dialog::settings_dialog(std::shared_ptr<gui_settings> gui_settings, std
 	m_emu_settings->EnhanceComboBox(ui->vulkansched, emu_settings_type::VulkanAsyncSchedulerDriver);
 	SubscribeTooltip(ui->gb_vulkansched, tooltips.settings.vulkan_async_scheduler);
 
+	// Custom control that simplifies operation of two independent variables. Can probably be done better but this works.
+	ui->zcullPrecisionMode->addItems({"Precise (Default)", "Approximate (Fast)", "Relaxed (Fastest)"});
+	const auto is_relaxed_zcull = m_emu_settings->GetSetting(emu_settings_type::RelaxedZCULL) == "true";
+	const auto is_precise_zcull = m_emu_settings->GetSetting(emu_settings_type::PreciseZCULL) == "true";
+	if (is_relaxed_zcull)
+	{
+		ui->zcullPrecisionMode->setCurrentIndex(2);
+	}
+	else if (is_precise_zcull)
+	{
+		ui->zcullPrecisionMode->setCurrentIndex(0);
+	}
+	else
+	{
+		ui->zcullPrecisionMode->setCurrentIndex(1);
+	}
+	connect(ui->zcullPrecisionMode, QOverload<int>::of(&QComboBox::currentIndexChanged), [this](int index)
+	{
+		bool relaxed = false, precise = false;
+
+		switch (index)
+		{
+		case 0:
+			precise = true; break;
+		case 1:
+			break;
+		case 2:
+			relaxed = true; break;
+		default:
+			fmt::throw_exception("Unexpected selection");
+		}
+
+		m_emu_settings->SetSetting(emu_settings_type::RelaxedZCULL, relaxed ? "true" : "false");
+		m_emu_settings->SetSetting(emu_settings_type::PreciseZCULL, precise ? "true" : "false");
```

Comment:
you might want to also set this once after you initialize the combobox, in order to ensure only one is selected at all times.

Reply:
Not a constraint, it is legal to select both but not exposed via GUI.

Reply:
What I mean is the following:
Consider you changed both values to true to test both at the same time.
Now you go to the settings and change some stuff.
You look at the ZCULL option and it says "Relaxed (fastest)", so you say mkay and save.

But now both values are still set to true because the GUI never changed them.

You're basically dropping consistency for the sake of the convenience for testers.

Reply:
This is a damned if I do, damned if I don't situation it seems. I did omit that by design, I do not want the GUI to silently overwrite my config options. The only way out of that would be to have even more options or have a special "Custom" entry in cases like that, but I really do not want to touch Qt code more than I have to.
In this case, relaxed + precise is still going to be behave correctly, and anyone with enough knowledge to set this manually via config.yml should know better than to rely on the dropdown to untangle any mess they create for themselves.

Reply:
Just wanted to make sure we're on the same page

Reply:
My reasoning is this - if I'm testing both options and I run into a problem and want to try a second or third GPU from another manufacturer, I should be able to do so in the GUI and continue my testing since I did not touch the setting. Since no casual user should be poking around in there, it should be ok to do so.

Reply:
Btw, relaxed will be faster than approximate even if you accidentally left the precise flag on. Since relaxed mode enables a different transport mechanism and allows leveraging GPU conditional rendering path, we can still skip most of the work associated with waiting for occlusion reports. This means even though precise requires more time for the GPU to process, it is keeping things GPU-side making the overhead minimal.